### PR TITLE
administration: add user management panel

### PR DESCRIPTION
**Pull request by TLGINO**
* added entry-points for invenio-users-resources
 * added dummy panel data + dummy html templates
 * closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2232

### CODE REVIEW

**Comment by TLGINO** [comment]
Current UI
image

Disclaimer: clicking on the "Users" link crashes (backend not implemented)

File: setup.cfg [hunk 1]
```diff
@@ -63,7 +64,11 @@ invenio_search.mappings =
     groups = invenio_users_resources.records.mappings
 invenio_i18n.translations =
     messages = invenio_users_resources
-
+invenio_administration.views =
+    invenio_users_resources_users_list = invenio_users_resources.administration.views.users:UsersListView
+    invenio_users_resources_users_edit = invenio_users_resources.administration.views.users:UsersEditView
+    invenio_users_resources_users_create = invenio_users_resources.administration.views.users:UsersCreateView
+    invenio_users_resources_details = invenio_users_resources.administration.views.users:UsersDetailView
```
Comment (ppanero) [inline setup.cfg:71]: minor: consistency
```suggestion
    invenio_users_resources_users_details = invenio_users_resources.administration.views.users:UsersDetailView
```

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-search.html
Comment (ppanero) [inline invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-search.html:1]: assuming this template will be further implemented after the integration with the backend?

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-details.html
Comment (ppanero) [inline invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-details.html:1]: assuming this template will be further implemented after the integration with the backend?

File: invenio_users_resources/administration/views/users.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    AdminResourceEditView,
+    AdminResourceListView,
+)
+from invenio_i18n import lazy_gettext as _
+
+
+class UsersListView(AdminResourceListView):
+    """Configuration for users sets list view."""
+
+    api_endpoint = "/users/sets"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:22]: [REST API reference](https://inveniordm.docs.cern.ch/reference/rest_api_users/)
```suggestion
    api_endpoint = "/users"
```

File: invenio_users_resources/administration/views/users.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    name = "USERS"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:23]: To be consistent with `Pages` and `Banners`
```suggestion
    name = "Users"
```

File: invenio_users_resources/administration/views/users.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    resource_config = "users_resource"
+    search_request_headers = {"Accept": "application/json"}
+    title = "USERS Sets"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:26]: _Sets_ is a specific term used on OAI-PMH

```suggestion
    title = "Users"
```

File: invenio_users_resources/administration/views/users.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    category = "User management"
+    pid_path = "id"
+    icon = "exchange"
+    template = "invenio_users_resources/users-search.html"
+
+    display_search = True
+    display_delete = True
+    display_edit = True
+
+    item_field_list = {
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:36]: The columns should match (when possible) the current ones. Maybe add if it's public or not (and other preferences/profile info)

Screenshot 2023-06-21 at 14 36 41

File: invenio_users_resources/administration/views/users.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    """Configuration for users sets edit view."""
+
+    name = "users_edit"
+    url = "/users/<pid_value>/edit"
+    resource_config = "users_resource"
+    pid_path = "id"
+    api_endpoint = "/users/sets"
+    title = "Edit USERS set"
+
+    list_view_name = "USERS"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:62]:
```suggestion
    list_view_name = "users"
```

File: invenio_users_resources/administration/views/users.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+
+class UsersCreateView(AdminResourceCreateView):
+    """Configuration for users sets create view."""
+
+    name = "users_create"
+    url = "/users/create"
+    resource_config = "users_resource"
+    pid_path = "id"
+    api_endpoint = "/users/sets"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:99]:
```suggestion
    api_endpoint = "/users"
```

File: invenio_users_resources/administration/views/users.py [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    title = "Create USERS set"
+
+    list_view_name = "USERS"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:102]:
```suggestion
    list_view_name = "users"
```

File: invenio_users_resources/administration/views/users.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+    form_fields = {
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:104]: same as before for the form

File: invenio_users_resources/administration/views/users.py [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+            ),
+        },
+    }
+
+
+class UsersDetailView(AdminResourceDetailView):
+    """Configuration for users sets detail view."""
+
+    url = "/users/<pid_value>"
+    api_endpoint = "/users/sets"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:133]:
```suggestion
    api_endpoint = "/users/<pid_value>"
```

File: invenio_users_resources/administration/views/users.py [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    search_request_headers = {"Accept": "application/json"}
+    name = "USERS details"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:135]: general comment on `name` if it is not a user facing attribute it shoudl be lowercased and snakecased (i.e. no spaces, but `_`). It applies to all views' name
```suggestion
    name = "User details"
```

File: invenio_users_resources/administration/views/users.py [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+    resource_config = "users_resource"
+    title = "USERS Details"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:137]:
```suggestion
    title = "User Details"
```

File: invenio_users_resources/administration/views/users.py [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+    template = "invenio_users_resources/users-details.html"
+    display_delete = True
+    display_edit = True
+
+    list_view_name = "USERS"
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:143]:
```suggestion
    list_view_name = "users"
```

File: invenio_users_resources/administration/views/users.py [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+    pid_path = "id"
+
+    item_field_list = {
```
Comment (ppanero) [inline invenio_users_resources/administration/views/users.py:146]: Same as before with the form fields

**Review by ppanero — COMMENTED** [review]
Nice initial job! A few comments, mostly on consistency and naming.

Extra:
- Test not passing on MySQL (what was the outcome @kpsherva on removing it from the matrix)
- Needs rebasing (conflicts)

Reviewed with @rekt-hard

**Comment by kpsherva** [comment]
implementation in : https://github.com/inveniosoftware/invenio-users-resources/pull/82

File: invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-details.html
Comment (psaiz) [reply invenio_users_resources/templates/semantic-ui/invenio_users_resources/users-details.html:1]: It depends on th edefault template. No need for an extra one

File: invenio_users_resources/administration/views/users.py (hunk 2 above)
Comment (psaiz) [reply invenio_users_resources/administration/views/users.py:22]: Done

File: invenio_users_resources/administration/views/users.py (hunk 10 above)
Comment (psaiz) [reply invenio_users_resources/administration/views/users.py:133]: Done

File: invenio_users_resources/administration/views/users.py (hunk 11 above)
Comment (psaiz) [reply invenio_users_resources/administration/views/users.py:135]: Done

**Pull request closed without merging**